### ISSUE TITLE
Package Manager OverflowException: Value too large or too small.

### ISSUE THREAD

**Issue by ericraue** [modesttree/Projeny#10]
When I open the Projeny -> Package Manager dialog the following error appears. This happens when I open the CubeMover-Windows project and also a project I created from scratch from the command line.

Error message: http://i.imgur.com/Se7Ms0M.png

Here's what it says in the console.
```
OverflowException: Value too large or too small.
System.Int32.Parse (System.String s, NumberStyles style, IFormatProvider provider) (at /Users/builduser/buildslave/mono-runtime-and-classlibs/build/mcs/class/corlib/System/Int32.cs:640)
System.ComponentModel.Int32Converter.ConvertFromString (System.String value, System.Globalization.NumberFormatInfo format)
System.ComponentModel.BaseNumberConverter.ConvertFrom (ITypeDescriptorContext context, System.Globalization.CultureInfo culture, System.Object value)
Rethrow as Exception: 3968277541 is not a valid value for Int32.
System.ComponentModel.BaseNumberConverter.ConvertFrom (ITypeDescriptorContext context, System.Globalization.CultureInfo culture, System.Object value)
YamlDotNet.Serialization.Utilities.TypeConverter.ChangeType (System.Object value, System.Type destinationType, System.Globalization.CultureInfo culture)
YamlDotNet.Serialization.Utilities.TypeConverter.ChangeType (System.Object value, System.Type destinationType, System.Globalization.CultureInfo culture)
YamlDotNet.Serialization.Utilities.TypeConverter.ChangeType (System.Object value, System.Type destinationType)
YamlDotNet.Serialization.NodeDeserializers.ScalarNodeDeserializer.YamlDotNet.Serialization.INodeDeserializer.Deserialize (YamlDotNet.Core.EventReader reader, System.Type expectedType, System.Func`3 nestedObjectDeserializer, System.Object& value)
YamlDotNet.Serialization.ValueDeserializers.NodeValueDeserializer.DeserializeValue (YamlDotNet.Core.EventReader reader, System.Type expectedType, YamlDotNet.Serialization.Utilities.SerializerState state, IValueDeserializer nestedObjectDeserializer)
Rethrow as YamlException: (Line: 25, Col: 17, Idx: 332) - (Line: 25, Col: 27, Idx: 342): Exception during deserialization
YamlDotNet.Serialization.ValueDeserializers.NodeValueDeserializer.DeserializeValue (YamlDotNet.Core.EventReader reader, System.Type expectedType, YamlDotNet.Serialization.Utilities.SerializerState state, IValueDeserializer nestedObjectDeserializer)
YamlDotNet.Serialization.ValueDeserializers.AliasValueDeserializer.DeserializeValue (YamlDotNet.Core.EventReader reader, System.Type expectedType, YamlDotNet.Serialization.Utilities.SerializerState state, IValueDeserializer nestedObjectDeserializer)
YamlDotNet.Serialization.ValueDeserializers.NodeValueDeserializer+<>c__DisplayClass2.<DeserializeValue>b__0 (YamlDotNet.Core.EventReader r, System.Type t)
YamlDotNet.Serialization.NodeDeserializers.ObjectNodeDeserializer.YamlDotNet.Serialization.INodeDeserializer.Deserialize (YamlDotNet.Core.EventReader reader, System.Type expectedType, System.Func`3 nestedObjectDeserializer, System.Object& value)
YamlDotNet.Serialization.ValueDeserializers.NodeValueDeserializer.DeserializeValue (YamlDotNet.Core.EventReader reader, System.Type expectedType, YamlDotNet.Serialization.Utilities.SerializerState state, IValueDeserializer nestedObjectDeserializer)
Rethrow as CoRoutineException: Coroutine Object Trace: PmWindowInitializer -> PmReleasesHandler -> PrjCommandHandler
Projeny.Internal.CoRoutine.Pump ()
Projeny.Internal.AsyncProcessor.AdvanceFrameAll ()
UnityEngine.Debug:LogException(Exception)
Projeny.Internal.Log:ErrorException(Exception)
Projeny.Internal.PmViewErrorHandler:OnAsyncException(Exception)
Projeny.Internal.AsyncProcessor:AdvanceFrameAll()
Projeny.Internal.AsyncProcessor:Tick()
Projeny.Internal.PmCompositionRoot:Update()
Projeny.Internal.PmWindow:Update()
UnityEditor.EditorApplication:Internal_CallUpdateFunctions()
```

Unity 5.3.2p1 (also tried on 5.3.0f4)
Python 3.5.1 64 bit

Tried compiling the Projeny source but the result is the same. Needed to install pywin32-220 and PyYAML-3.11 to compile.

**Comment by svermeulen** [comment]
I can't reproduce this, but from the error it looks like the issue is occurring when Projeny scans your asset store cache.  This cache can be found at `C:\Users\YOURUSERNAME\AppData\Roaming\Unity\` in the folders `Asset Store` and `Asset Store-5.x`.

My best guess is that one of the unitypackage files in those folders has that number as its version number.  And I'm serializing the version number as an integer.  So if that's the case, the fix would be just to make it a long instead.

**Comment by svermeulen** [comment]
I tried changing it to use a long.  Can you try updating and then running it from source to see if that fixes it?

**Comment by ericraue** [comment]
Found the problematic package: [The Blacksmith: Environments](https://www.assetstore.unity3d.com/en/#!/content/39948). It's 3.8 GB which might have something to do with it.

I tried 24bd735 with no effect but I'm not entirely familiar with your build process yet.

**Comment by svermeulen** [comment]
Looks like that change 24bd735d155c005660ba2e5632c4ace376ac3cf5 does fix it.  I was able to reproduce it before the change and after getting the change it's gone.

**Comment by ericraue** [comment]
I thought it would, which is why I suspected my build process. My process was: Open Projeny/UnityPlugin/Projeny.sln, Clean Solution, Rebuild Solution. I also made a fresh copy of the files in the Projeny/Demo folder and initialized it with prj --init. My path includes C:\Projeny\Source\bin as mentioned in the installation instructions. I double checked to see if the file changes were there. What am I missing?

**Comment by svermeulen** [comment]
Hmm that's odd, what you did sounds correct.  I'll update the binary release with the fix and maybe that will help

**Comment by svermeulen** [comment]
One thing that occurred to me is to make sure you're compiling the solution in Release mode.  If not it might still be using the old dll before you got the fix

**Comment by svermeulen** [comment]
Closing this but if you find it's still not fixed in version 0.3 feel free to re-open

**Comment by ericraue** [comment]
I updated to the latest source and everything works. I think what might have happened is the YamlDotNet project didn't have the UnityEditor and UnityEngine references set correctly. I don't remember seeing warnings about that before but when I updated to latest they were there. Thanks for the help!
